Add more specific read/write examples in the data node entity page (#246)
Avaiga/taipy-doc: docs/manuals/core/entities/data-node-mgt.md

    data_node.read()
    ```

To write some data on the data node, like the output of a task, you can use the `DataNode.write()^` method. The

> method takes a data object (string, dictionary, lists, numpy arrays, pandas dataframes, etc. based on the storage type)

Reviewer: ```suggestion
method takes a data object (string, dictionary, lists, numpy arrays, pandas dataframes, etc. based on the data node type and its exposed type)
```

                "nb_sales": [1550, 2315, 1832]
            }
        )
        # or using a list of dictionaries
        data_node.write(
            [
                {"date": "12/24/2018", "nb_sales": 1550},
                {"date": "12/24/2018", "nb_sales": 2315},
                {"date": "12/24/2018", "nb_sales": 1832},
            ]
        )
        ```

    === "pandas dataframes"

        ```python
        data = pandas.DataFrame(
            [
                {"date": "12/24/2018", "nb_sales": 1550},
                {"date": "12/24/2018", "nb_sales": 2315},
                {"date": "12/24/2018", "nb_sales": 1832},
            ]
        )

        data_node.write(data)
        ```

When write a list or numpy array to Excel data node, the column name will be numbered from 1.
To write with custom column names, use the `ExcelDataNode.write_with_column_names()^` method.

!!! example "`ExcelDataNode.write_with_column_names()^` examples"

    ```python
    data_node.write(
        [
            ["12/24/2018", 1550],
            ["12/25/2018", 2315],
            ["12/26/2018", 1832],
        ],
        columns=["date", "nb_sales"]
    )
    ```

## SQL Table

> SQL table read/write example

Reviewer: todo

## SQL

> SQL read/write example

Reviewer: todo

!!! example "Read and write from a Mongo collection data node using default document class"

    ```python
    from taipy.core import DefaultCustomDocument

    data = [
        DefaultCustomDocument(date="12/24/2018", nb_sales=1550),
        DefaultCustomDocument(date="12/25/2018", nb_sales=2315),
        DefaultCustomDocument(date="12/26/2018", nb_sales=1832),
    ]
    data_node.write(data)
    ```

    will write 3 documents to MongoDB:

    ```json
    [
        {"_id": ObjectId("634cd1b3383279c68cee1c21"), "date": "12/24/2018", "nb_sales": 1550},
        {"_id": ObjectId("634cd1b3383279c68cee1c22"), "date": "12/25/2018", "nb_sales": 2315},
        {"_id": ObjectId("634cd1b3383279c68cee1c23"), "date": "12/26/2018", "nb_sales": 1832},
    ]
    ```

    The read method will return a list of DefaultCustomDocument objects, including "_id" attribute.

You can also specify custom document class to handle specific attribute, encode and decode data when reading and writing to the Mongo collection.

Check out [Mongo collection Data Node configuration](../config/data-node-config.md#mongo-collection) for more details on how to config a custom
document class.

## In memory

> In memory read / write example

Reviewer: todo.
for in memory data nodes, can you add a warning node to recall that in memory should only be used in development mode.

## Generic

> Generic read / write example

Reviewer: todo

> # Filter read results

Reviewer: Can you check that the filter method is working for any kind of data node. I am not sure anymore at this point.

Author: For now, the filter method only work when the return type of read method is DataFrame, or List. So when the expose type is numpy, a custom class, or MongoCollectionDataNode, this will not work.

We'll need to specify supported data node, and update this section latter when the filter method is expanded.

document class.

## In memory

In memory read / write example

!!! Warning

    Since the data is stored in memory, it cannot be used in a multiprocess environment. (See
    [Job configuration](job-config.md#standalone) for more details).

## Generic

Generic read / write example

# Filter read results

It is also possible to partially read the contents of data nodes, which comes in handy when dealing with large amounts
of data.
This can be achieved by providing an operator, a Tuple of (*field_name*, *value*, *comparison_operator*),
or a list of operators to the `DataNode.filter()^` method:

```python linenums="1"
data_node.filter(
    [("field_name", 14, Operator.EQUAL), ("field_name", 10, Operator.EQUAL)],
    JoinOperator.OR
)
```

If a list of operators is provided, it is necessary to provide a join operator that will be
used to combine the filtered results from the operators.

It is also possible to use pandas style filtering:

```python linenums="1"
temp_data = data_node["field_name"]
temp_data[(temp_data == 14) | (temp_data == 10)]
```

!!! Warning

>     For now , the `DataNode.filter()^` method can only be used with data node that has the exposed type is pandas.DataFrame.

Reviewer: ```suggestion
    For now, the `DataNode.filter()^` method is only implemented for `CSVDataNode^`, `ExcelDataNode^`, `SQLTableDataNode^`, `SQLDataNode` with `"pandas"` as _**exposed_type**_ value.
```
